--- origsrc/db-4.0.14/dist/Makefile.in	2001-11-09 16:48:45.000000000 -0600
+++ src/db-4.0.14/dist/Makefile.in	2008-07-11 11:47:09.578125000 -0500
@@ -121,6 +121,7 @@
 # Tcl support is optional and requires shared libraries.
 ##################################################
 TCFLAGS=	@TCFLAGS@
+TSOLINK=		@MAKEFILE_TSOLINK@
 LIBTSO_LIBS=	@LIBTSO_LIBS@
 libtso_base=	libdb_tcl
 libtso=		$(libtso_base)-$(SOVERSION).@SOSUFFIX@
@@ -297,7 +298,7 @@
 	    -o $@ $(JAVA_OBJS) $(C_OBJS) $(LIBJSO_LIBS)
 
 $(libtso_target): $(TCL_OBJS) $(C_OBJS)
-	$(SOLINK) $(SOFLAGS) $(LDFLAGS) \
+	$(TSOLINK) $(SOFLAGS) $(LDFLAGS) \
 	    -o $@ $(TCL_OBJS) $(C_OBJS) $(LIBTSO_LIBS)
 
 $(libxso_target): $(CXX_OBJS) $(C_OBJS)
--- origsrc/db-4.0.14/dist/aclocal/tcl.ac	2001-11-11 12:01:34.000000000 -0600
+++ src/db-4.0.14/dist/aclocal/tcl.ac	2008-07-11 11:47:09.593750000 -0500
@@ -101,7 +101,7 @@
 	# load time. [#4843]
 	#
 	case "$host_os" in
-	aix*)
+	aix*|cygwin*)
 		LIBTSO_LIBS="$LIBTSO_LIBS $TCL_LIB_SPEC $TCL_LIB_FLAG";;
 	esac
 
--- origsrc/db-4.0.14/dist/aclocal_java/ac_jni_include_dirs.ac	2001-09-19 08:58:33.000000000 -0500
+++ src/db-4.0.14/dist/aclocal_java/ac_jni_include_dirs.ac	2008-07-11 13:34:43.203125000 -0500
@@ -35,7 +35,7 @@
 JNI_INCLUDE_DIRS=""
 
 test "x$JAVAC" = x && AC_MSG_ERROR(['$JAVAC' undefined])
-AC_PATH_PROG(_ACJNI_JAVAC, $JAVAC, no)
+AC_PATH_PROG(_ACJNI_JAVAC, $JAVAC, $JAVAC)
 test "x$_ACJNI_JAVAC" = xno && AC_MSG_ERROR([$JAVAC could not be found in path])
 
 _ACJNI_FOLLOW_SYMLINKS("$_ACJNI_JAVAC")
--- origsrc/db-4.0.14/dist/configure.ac	2001-11-16 10:15:01.000000000 -0600
+++ src/db-4.0.14/dist/configure.ac	2008-07-11 13:35:17.156250000 -0500
@@ -53,6 +53,7 @@
 AC_SUBST(MAKEFILE_CXXLINK)
 AC_SUBST(MAKEFILE_SOLINK)
 AC_SUBST(MAKEFILE_XSOLINK)
+AC_SUBST(MAKEFILE_TSOLINK)
 AC_SUBST(POSTLINK)
 AC_SUBST(RPC_CLIENT_OBJS)
 AC_SUBST(RPM_POST_INSTALL)
@@ -263,8 +264,8 @@
 versuffix=""
 release=""
 libname=libfoo
-eval library_names=\"$library_names_spec\"
-SOSUFFIX=`echo "$library_names" | sed -e 's/.*\.\([[a-zA-Z0-9_]]*\).*/\1/'`
+eval library_names=\"$shrext_cmds\"
+SOSUFFIX=`echo "$library_names"`
 AC_MSG_RESULT($SOSUFFIX)
 
 if test "$SOSUFFIX" = '' ; then
@@ -279,11 +280,19 @@
 
 MAKEFILE_CC="\$(LIBTOOL) --mode=compile ${MAKEFILE_CC}"
 MAKEFILE_SOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -avoid-version"
+MAKEFILE_TSOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK} -module -avoid-version"
 MAKEFILE_CCLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CCLINK}"
 MAKEFILE_CXX="\$(LIBTOOL) --mode=compile ${MAKEFILE_CXX}"
 MAKEFILE_XSOLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK} -avoid-version"
 MAKEFILE_CXXLINK="\$(LIBTOOL) --mode=link ${MAKEFILE_CXXLINK}"
 
+case "$host_os" in
+	cygwin*)
+		MAKEFILE_SOLINK="$MAKEFILE_SOLINK -no-undefined"
+		MAKEFILE_XSOLINK="$MAKEFILE_XSOLINK -no-undefined"
+		MAKEFILE_TSOLINK="$MAKEFILE_TSOLINK -no-undefined"
+esac
+
 # Configure for shared libraries, static libraries, or both.  If both are
 # configured, build the utilities and example programs with shared versions.
 #
--- origsrc/db-4.0.14/lock/lock_region.c	2001-11-05 09:24:05.000000000 -0600
+++ src/db-4.0.14/lock/lock_region.c	2008-07-11 11:47:09.625000000 -0500
@@ -21,6 +21,10 @@
 #include "db_shash.h"
 #include "lock.h"
 
+#ifdef __lock_init
+#undef __lock_init
+#endif
+
 static int  __lock_init __P((DB_ENV *, DB_LOCKTAB *));
 static size_t
 	    __lock_region_size __P((DB_ENV *));
